src/inventory_ui.cpp

```diff
@@ -401,14 +454,329 @@ void inventory_selector::print_right_column() const
+                auto const &nc_text = get_item_hp_bar(it.damage);
+                mvwprintz(w_inv, curr_line + 1, 6 + (row_i * 25), nc_text.second, nc_text.first.c_str());
+            }
+            
+            mvwprintz(w_inv, curr_line + 1, 10 + (row_i * 25), fill, it.get_mod_string().c_str()); // Show mod count (if any)
+        }
+        
+        std::string temp[2];
+        std::string *name_lines = split_name(item_name, temp);
+        
+        if(curr_line >= 1 && curr_line + 2 < TERMY) {  // Third line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 2, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 2, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 2, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
+            
+            mvwprintz(w_inv, curr_line + 2, 2 + (row_i * 25), name_color, name_lines[0].c_str()); // Print line one of name
+        }
+        
+        if(curr_line >= 0 && curr_line + 3 < TERMY) {  // Fourth line
+            // Cell Border Stuff
+            if(row_i == 0) {
+                mvwputch(w_inv, curr_line + 3, 1, c_white, LINE_XOXO);
+            }
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 3, x + (row_i * 25), fill, ' ');
+            }
+            mvwputch(w_inv, curr_line + 3, 26 + (row_i * 25), c_white, LINE_XOXO);
+            // End Cell Border Stuff
+            
+            mvwprintz(w_inv, curr_line + 3, 4 + (row_i * 25), name_color, name_lines[1].c_str());
+            
+            std::string tagtext = it.get_item_tags(false);
+            
+            if(tagtext.length() + name_lines[1].length() > 21) { // Collision between tags and second name line
+                mvwprintz(w_inv, curr_line + 3, 23 - tagtext.length() + (row_i * 25), fill, "...%s", tagtext.c_str());
+            } else {
+                mvwprintz(w_inv, curr_line + 3, 26 - tagtext.length() + (row_i * 25), fill, tagtext.c_str());
+            }
+        }
+        
+        if(curr_line >= -1 && curr_line + 4 < TERMY) {  // Bottom line
+            mvwputch(w_inv, curr_line + 4, 1 + (row_i * 25), c_white, (row_i == 0) ? LINE_XXOO : LINE_XXOX);
+            for(char x = 2; x < 26; x++) {
+                mvwputch(w_inv, curr_line + 4, x + (row_i * 25), c_white, LINE_OXOX);
+            }
+            mvwputch(w_inv, curr_line + 4, 26 + (row_i * 25), c_white, LINE_XOOX);
+        }
+        
+        if(++row_i >= row_width) {
+            row_i = 0;
+            curr_line += 4;
+            cat_continued = true;
+        }
+    }
+    
+    // Bit of spacer between worn and carried...
+    curr_line += 2;
+    total_height += 2;
+    
+    // Then the carried stuff
+    for(size_t a = 0; a < inv.size(); a++) {
```

Comment:
Is the body of this loop a direct copy of the body of the loop above? Why would you do such a thing? Move the code into a function and call it from within both loops.